flake.nix

```diff
@@ -99,6 +101,31 @@
             x11-gnome = runVM ./nix/vm/x11-gnome.nix;
             x11-plasma6 = runVM ./nix/vm/x11-plasma6.nix;
             x11-xfce = runVM ./nix/vm/x11-xfce.nix;
+            make-cache = let
+              program = pkgs-stable.writeShellScript "make-cache" ''
+                set -e
+                mkdir -p blob
+                WORK_DIR=$(mktemp -d)
+                if [[ ! "$WORK_DIR" || ! -d "$WORK_DIR" ]]; then
+                  echo "Could not create temp dir"
+                  exit 1
+                fi
+                function cleanup {
+                  rm -rf "$WORK_DIR"
+                }
+                trap cleanup EXIT
+                ln -s "${self.packages.${system}.deps}" "$WORK_DIR/cache"
+                pushd "$WORK_DIR/cache" > /dev/null
+                tar --create --file "$WORK_DIR/temp.tar.gz" */*
+                popd > /dev/null
+                HASH=$(sha256sum "$WORK_DIR/temp.tar.gz" | awk "NR == 1 { print \$1 }; NR>1 { exit }")
+                mv "$WORK_DIR/temp.tar.gz" "blob/$HASH.tar.gz"
+                ln -sf "$HASH.tar.gz" "blob/ghostty-cache-${version}-${revision}.tar.gz"
```

Comment:
I haven't run this yet, but if I'm reading this correctly it's named two things. I think I'd prefer to have this named `zig-cache.tar.gz` and let the caller deal with renaming it to what they see fit, its just so much easier with scripting that way.

Reply:
It's done that way so that there isn't a 500+ MB file for each git commit. By naming it after the hash of the contents and then adding a symlink from the git commit space usage is minimized.

Reply:
Sorry I mean just outputting one file and letting the release automation deal with what it wants to do otherwise. For uploading to R2 I think we need the physical file anyways, I'm not sure if the CI script (that we don't own/maintain) follows symlinks.